src/openforms/analytics_tools/tests/test_models.py

```diff
@@ -68,11 +77,91 @@ def test_analytics_tool_properly_disabled(self):
 
         for csp in self.json_csp:
             with self.subTest("Test deletion of CSP"):
-                self.assertFalse(CSPSetting.objects.filter(value=csp["value"]).exists())
+                self.assertFalse(
+                    CSPSetting.objects.filter(
+                        value=csp["value"], directive=csp["directive"]
+                    ).exists()
+                )
 
-    def test_analytics_tool_enabled_but_related_fields_are_not(self):
+    def test_google_analytics_enabled_but_related_fields_are_not(self):
         self.config.enable_google_analytics = True
         self.config.gtm_code = ""
+
+        with self.assertRaises(ValidationError):
+            self.config.clean()
+
+
+@override_settings(
+    SOLO_CACHE=None,
+)
+class MatomoTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.matomo_url = "https://example.com"
+        cls.matomo_site_id = 1234
+        string_replacements_list = [
+            ("SITE_ID", cls.matomo_site_id),
+            ("SITE_URL", cls.matomo_url),
+            (
+                "DOMAIN_HASH",
+                lambda cookie: get_domain_hash(
+                    settings.ALLOWED_HOSTS[0], cookie_path=cookie["path"]
+                ),
+            ),
+        ]
+        cls.json_cookies = get_cookies(
+            analytics_tool=AnalyticsTools.matomo,
+            string_replacement_list=string_replacements_list,
+        )
+
+        cls.json_csp = get_csp(
+            analytics_tool=AnalyticsTools.matomo,
+            string_replacement_list=string_replacements_list,
+        )
```

Comment:
I think I'd avoid re-usin the implementation code here, and instead just use a hardcoded list of cookies/csp directives in the test for what you'd expect.

That means that if it turns out something is missing, we can update the test and if someone messes with the JSON files without knowing what they're doing, we have tests that check for the correct lists of cookies/CSP directives.

So in short - please hardcode this in the test rather than using the utility functions.